airflow/contrib/sensors/triggered_dagrun_sensor.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains the TriggeredDagSensor which allows for checking the status
+of triggered DAG runs."""
+from airflow.exceptions import AirflowException
+from airflow.utils import db
+from airflow.utils.state import State
+from airflow.utils.decorators import apply_defaults
+from airflow.utils.trigger_rule import TriggerRule
+from airflow.models import DagRun
+from airflow.sensors.base_sensor_operator import BaseSensorOperator
+
+
+class TriggeredDagRunSensor(BaseSensorOperator):
+    """
+    Waits for triggered DAG run(s) to complete and checks status.
+
+    :param trigger_task_id: The id of the task that triggered the dags
+        and returns a list of dagrun ids to monitor.
+    :type trigger_task_id: str
+    :param sensor_rule: criteria for success after dagruns complete.
+        Default is ``TriggerRule.ONE_SUCCESS``
+    :type sensor_rule: str
+    """
+    @apply_defaults
+    def __init__(
+            self,
+            trigger_task_id,
+            sensor_rule=None,
+            *args, **kwargs):
+        super(TriggeredDagRunSensor, self).__init__(*args, **kwargs)
+        self.sensor_rule = sensor_rule or TriggerRule.ONE_SUCCESS
+        self.trigger_task_id = trigger_task_id
+
+    def poke(self, context):
+        with db.create_session() as session:
+            runcount = 0
+            ti = context['ti']
+            dagrun_ids = ti.xcom_pull(task_ids=self.trigger_task_id)
```

Comment:
How about we the dagrun_ids a templated parameter instead?

Reply:
I'm unclear how that would work with the case I have in the triggered_dagrun_sensor_test.py since I haven't used templated parameters much:
```
def triggerchild():
        dr = trigger_dag(TEST_DAG_CHILD)
        return [dr.run_id]
t0 = PythonOperator(python_callable=triggerchild,
                        task_id='trigger_child')
t1 = TriggeredDagRunSensor(
        task_id='sense_child',
        trigger_task_id='trigger_child',
        timeout=15,
        poke_interval=1,
    )
```
The issue is that usually the triggerchild() has a loop that triggers several dagruns and returns a list of their ids.  If this works with a templated parameter, then it's fine by me.